firedrake/projection.py

```diff
@@ -97,3 +97,41 @@ def project(v, V, bcs=None, mesh=None,
            solver_parameters=solver_parameters,
            form_compiler_parameters=form_compiler_parameters)
     return ret
+
+
+class Projector(object):
+    """
+    A projector projects a UFL expression into a function space
+    and places the result in a function from that function space,
+    allowing the solver to be reused.
+
+    :arg v: the :class:`.Expression`, :class:`ufl.Expr` or
+         :class:`.Function` to project
+    :arg v_out: :class:`.Function` to put the result in
+    :arg solver_parameters: parameters to pass to the solver used when
+         projecting.    
+    """
+
+    def __init__(self, v, v_out, solver_parameters = None):
```

Comment:
I would maybe be tempted to use `LinearSolver` object instead of a `LVS`.  That way you can have one projector object per space you want to project into:

```
def __init__(self, V, solver_parameters=None):
    p = TestFunction(V)
    q = TrialFunction(V)

    A = assemble(inner(p, q)*dx)

    self.solver = LinearSolver(A, solver_parameters=solver_parameters)
    self.p = p

def project(self, v, v_out):
    self.solver.solve(v_out, assemble(inner(v, self.p)*dx))
```

Although maybe this is worse because now whenever you project you build a new form.